### ISSUE TITLE
config_write can add duplicate entry

### ISSUE THREAD

**Issue by rcorre** [libgit2/libgit2#3072]
PR #3064 did not completely solve the problems with duplicate config headers:

Given a config file that looks like:
```
[foo]
    a = b
[bar]
    a = b
[foo]
    c = old
```

`git_config_set_string(cfg, "foo.c", "new")` will create:

```
[foo]
    a = b
    c = new
[bar]
    a = b
[foo]
    c = old
```

When fetching a key with multiple entries, the last one is taken, so in this case, the value of "foo.c" is still considered to be "old"

**Comment by ethomson** [comment]
I'm looking at this now.

**Comment by rcorre** [comment]
Wow that was fast. There's a unittest in that PR if you're interested.

**Comment by ethomson** [comment]
:sparkles:  I love that you write unit tests when you find a bug.

**Comment by ethomson** [comment]
Closed via #3079
